### PR TITLE
Test unit added for `query` method for `TemplateFunctions` module

### PR DESCRIPTION
**Pull request by nogazr**
#### Test Plan
- [x] Verify the tests are passing

### CODE REVIEW

File: spec/consult_spec.rb [hunk 1]
```diff
@@ -19,12 +19,17 @@
...
 
-    # Verify text templates rendered correctly
-    %w[elements.txt more_elements.txt consul_elements.txt more_consul_elements.txt multi_pass.txt].each do |template|
-      expect(FileUtils.compare_file("spec/support/expected/#{template}", "spec/support/rendered/#{template}")).to be true
+    it 'renders without error' do
+      expect { Consult.render! }.to_not raise_exception
+    end
+
+    it 'renders templates correctly' do
+      %w[elements.txt more_elements.txt consul_elements.txt more_consul_elements.txt multi_pass.txt query_element.yml].each do |template|
```
Comment (nogazr) [inline spec/consult_spec.rb:30]: query_element.yml was added, this one uses the query method to be renderer

File: spec/support/consul-query.json [hunk 2]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (nogazr) [inline spec/support/consul-query.json:1]: This query was required to test the method query for TemplateFunctions

File: spec/support/populate_consul.sh [hunk 3]
```diff
@@ -31,3 +31,14 @@ curl \
     -H "Content-Type: application/x-www-form-urlencoded; charset=utf-8" \
     --data $'Aziz! <%= vars[:aziz] %> 🙄\n' \
     http://localhost:8500/v1/kv/templates/elements/aziz
+
+curl \
```
Comment (nogazr) [inline spec/support/populate_consul.sh:35]: We upload a template to use the method query

File: spec/support/populate_consul.sh [hunk 4]
```diff
@@ -31,3 +31,14 @@ curl \
...
+    --request PUT \
+    -H "Content-Type: application/x-www-form-urlencoded; charset=utf-8" \
+    --data-binary @spec/support/templates/query-test.yml.erb \
+    http://localhost:8500/v1/kv/templates/db/db1
+
+curl \
```
Comment (nogazr) [inline spec/support/populate_consul.sh:41]: With this curl we create the query on consul

File: spec/support/templates/query-test.yml.erb [hunk 5]
```diff
@@ -0,0 +1,5 @@
+configuration:
```
Comment (nogazr) [inline spec/support/templates/query-test.yml.erb:1]: With this template we test the method query for TemplateFunctions

**Review by aharpervc — APPROVED** [review]
LGTM, love it when we add more tests.

---

To be clear for others, we believe that https://github.com/veracross/consult/pull/37 revealed a bug that is solved by changing the default args from nil to `{}`, and we also now have better test coverage for that behavior. @nogazr correct?

**Review by ronnietaylor — APPROVED** [review]
These changes seem good. Prior to merging will you point to this branch in your Rails upgrade pr and make sure you can deploy to staging as a way to validate that template rendering is working in that context?

**Comment by nogazr** [comment]
> LGTM, love it when we add more tests.
> 
> To be clear for others, we believe that #37 revealed a bug that is solved by changing the default args from nil to `{}`, and we also now have better test coverage for that behavior. @nogazr correct?

Yes, that's right! @aharpervc

**Pull request merged**